xds/internal/xdsclient/xdsresource/unmarshal_rds.go

```diff
@@ -83,9 +83,13 @@ func unmarshalRouteConfigResource(r *anypb.Any, logger *grpclog.PrefixLogger) (s
 // we are looking for.
 func generateRDSUpdateFromRouteConfiguration(rc *v3routepb.RouteConfiguration, logger *grpclog.PrefixLogger, v2 bool) (RouteConfigUpdate, error) {
 	vhs := make([]*VirtualHost, 0, len(rc.GetVirtualHosts()))
-	csps, err := processClusterSpecifierPlugins(rc.ClusterSpecifierPlugins)
-	if err != nil {
-		return RouteConfigUpdate{}, fmt.Errorf("received route is invalid %v", err)
+	csps := make(map[string]clusterspecifier.BalancerConfig)
+	var err error
```

Comment:
Declare this inside the `if` just below please.

Reply:
Done.